agents/build.sh

```diff
@@ -0,0 +1,50 @@
+#!/usr/bin/env bash
+#
+# Copyright 2017 The Kubeflow Examples Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+get_project_id() {
+  local project="${DEVSHELL_PROJECT_ID:-}"
+  if [[ -z "$project" ]]; then
+    project=$(gcloud config get-value project 2> /dev/null)
+  fi
+  if [[ -z "$project" ]]; then
+    >&2 echo "No default project was found."
+    >&2 echo "Please set a project by typing:"
+    >&2 echo "gcloud config set project YOUR-PROJECT-NAME"
+  fi
+  echo "$project"
+}
+
+SCRIPT_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+
+IMAGE_BASE_NAME=agents
+TENSORFLOW_IMAGE_TAG=1.4.1
+PROJECT_ID=$(get_project_id)
+SALT=`python -c 'import datetime; import uuid; now=datetime.datetime.now(); print(now.strftime("%m%d-%H%M") + "-" + uuid.uuid4().hex[0:4])'`
+
+VERSION_TAG=tf${TENSORFLOW_IMAGE_TAG}-${SALT}
+IMAGE_TAG=gcr.io/${PROJECT_ID}/${IMAGE_BASE_NAME}:${VERSION_TAG}
```

Comment:
@cwbeitel , @jlewi : Can we make images publicly available under a kubeflow project? Would save users having to go through the trouble of this build step.

We should be publishing to DockerHub (re: https://github.com/kubeflow/kubeflow/issues/211) as the primary source of images (as well as GCR, if we have a project handy).

Reply:
True it would be simpler to use if the example were contained in a publicly-accessible image. People would be able to either run the example from that image or make modifications and build their own.

Reply:
What is the image for?
In general users have to build their own Docker images so we should figure out how to make that CUJ work within cluster. 

I think the general solution is going to be by building docker images in cluster using Docker in Docker using Argo.
You can look at the Seldon example
https://github.com/SeldonIO/kubeflow-seldon-example

Reply:
That could probably be done in a follow up PR though.